Userland/Utilities/grep.cpp

```diff
@@ -56,6 +58,61 @@ static DeprecatedString escape_characters(StringView string, StringView characte
     return builder.to_deprecated_string();
 }
 
+static DeprecatedString& hostname()
+{
+    static DeprecatedString s_hostname;
+    if (s_hostname.is_null()) {
```

Comment:
the null state of DeprecatedString is particularly unpleasant, so it would be best to avoid it at all costs. We could either play games to try to initialize it on first entry, or just check the is_empty() state.

Reply:
Andrew probably means an immediately evaluated lambda by that, instead of relying of the default constructor of `DeprecatedString` generating the null state:

```c++
    static DeprecatedString s_hostname = [] {
        auto result = Core::System::gethostname();
        if (result.is_error())
            return "localhost";
        return result.release_value();
    };
```

(on first entry, the lambda is executed and the function-local static variable is initialized with its return value. On subsequent calls, the value is returned directly).

Reply:
right, but the easier thing to do would be to check is_empty() instead :)